File: examples/c++/eldriver.cxx
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2015 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/
#include <iostream>
#include <signal.h>

#include "eldriver.hpp"
#include "upm_utilities.h"

using namespace std;

int shouldRun = true;

void
sig_handler(int signo)
{
    if (signo == SIGINT)
        shouldRun = false;
}

int
main(int argc, char** argv)
{
    signal(SIGINT, sig_handler);

    //! [Interesting]
    // This was tested with the El Driver Module
    // Instantiate an El Driver on digital pin D2
    upm::ElDriver eldriver(2);

    bool lightState = true;

    while (shouldRun) {
        if (lightState)
            eldriver.on();
        else
            eldriver.off();
        lightState = !lightState;
        upm_delay(1);
    }

    //! [Interesting]
    eldriver.off();
    cout << "Exiting" << endl;

    return 0;
}

File: examples/javascript/eldriver.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2015 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var eldriver_lib = require("jsupm_eldriver");

// The was tested with the  El Driver Module
// Instantiate a  El Driver on digital pin D2
var eldriver_obj = new eldriver_lib.ElDriver(2);

var lightState = true;

var myInterval = setInterval(function()
{
	if (lightState)
		eldriver_obj.on();
	else
		eldriver_obj.off();
	lightState = !lightState;
}, 1000);

// When exiting: turn off EL wire, clear interval, and print message
process.on('SIGINT', function()
{
	clearInterval(myInterval);
	eldriver_obj.off();
	eldriver_obj = null;
	eldriver_lib.cleanUp();
	eldriver_lib = null;
	console.log("Exiting...");
	process.exit(0);
});
